Enable Erisc IRAM by default when persistent fabric is launched for Wormhole

SeanNijjar opened the pull request:
Note that this PR was originally intended to enable erisc IRAM support by default for all kernels. However, that was not possible due to performance regressions for some legacy CCLs where the overhead to copy to IRAM was an overall net loss in performance when compared to the instruction fetch speedups.

Therefore global default enablement is deferred until legacy CCLs are ported to run on fabric. For this PR, we instead only enable the IRAM feature on Erisc when building fabric. 

The overall flow for deciding if Erisc IRAM is enabled is then as follows:
- if user specifies `TT_METAL_ENABLE_ERISC_IRAM` flag, override all other behaviour with user choice
- else if/when fabric instantiated at device init time, enable IRAM
  - disable when torn down
- else Erisc iram disabled

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/16078141887
- [x] BH PC: https://github.com/tenstorrent/tt-metal/actions/runs/16081848208
- [x] T3K unit: https://github.com/tenstorrent/tt-metal/actions/runs/16078143816
- [x] T3K demo, frequent, nightly, model perf: https://github.com/tenstorrent/tt-metal/actions/runs/16081850565
  - demo failures same as main: https://github.com/tenstorrent/tt-metal/actions/runs/16062787249
  - frequent: https://github.com/tenstorrent/tt-metal/actions/runs/16133050434 (after testlist fixes)
- [x] TG: https://github.com/tenstorrent/tt-metal/actions/runs/15982260830
- [x] TG Stress: https://github.com/tenstorrent/tt-metal/actions/runs/15914430447
- [x] metal ubench: https://github.com/tenstorrent/tt-metal/actions/runs/15916997396
  - same fails as main 
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866679747
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866681192
- [x] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866686623

Closes #20340

daminakaTT commented on .github/workflows/tg-stress.yaml:
```diff
@@ -28,7 +28,7 @@ jobs:
         LOGURU_LEVEL: INFO
         LLAMA_DIR: /mnt/MLPerf/tt_dnn-models/llama/Llama3.1-70B-Instruct/
-        TT_METAL_ENABLE_ERISC_IRAM: 1
+        TT_METAL_WORKER_RINGBUFFER_SIZE: 122880
```
how does this relate to IRAM enablement?

SeanNijjar replied:
It doesn't this was a bad rebase conflict resolution. Fixed

SeanNijjar committed:
add IRAM disable. For

SeanNijjar committed:
Enable Erisc IRAM by default for device init fabric

hoist set iram enabled to earliest

some cleanup. cache the env var override for erisc iram enable so we can restore it during fabric teardown

SeanNijjar committed:
experiment

SeanNijjar committed:
experiment (force iram enable so it's consistent for whole workload run)

SeanNijjar committed:
experiment pt2

SeanNijjar committed:
force enable IRAM for test list with mixed fabric and non-fabric CCLs

sq

SeanNijjar committed:
update all tests to use IRAM

SeanNijjar committed:
Merge branch 'main' into snijjar/iram-on-by-default

SeanNijjar committed:
delete extra whitespace in run_tg_frequent_tests.sh
